hydra-node/test/Hydra/Chain/Direct/TimeHandleSpec.hs

```diff
@@ -3,15 +3,29 @@ module Hydra.Chain.Direct.TimeHandleSpec where
 import Hydra.Prelude hiding (label)
 import Test.Hydra.Prelude
 
-import Hydra.Chain.Direct.TimeHandle (TimeHandle (..))
+import Cardano.Ledger.Slot (SlotNo (SlotNo))
+import Cardano.Slotting.Time (SystemStart (SystemStart))
+import Data.Time.Clock.POSIX (posixSecondsToUTCTime)
+import Hydra.Chain.Direct.TimeHandle (TimeHandle (..), mkTimeHandle)
+import Hydra.Ledger.Cardano.Evaluate (eraHistoryWithHorizonAt)
 import Test.QuickCheck (counterexample, forAllBlind, property, (===))
 
 spec :: Spec
-spec =
+spec = do
   prop "can roundtrip currentPointInTime" $
     forAllBlind arbitrary $ \TimeHandle{currentPointInTime, slotToUTCTime, slotFromUTCTime} ->
       let onLeft err = property False & counterexample ("Conversion failed: " <> toString err)
        in either onLeft id $ do
             (slot, _) <- currentPointInTime
             res <- slotFromUTCTime =<< slotToUTCTime slot
             pure $ res === slot
+
+  it "should convert slot within latest/current era" $ do
+    let now = posixSecondsToUTCTime 13.4
+        systemStart = SystemStart $ posixSecondsToUTCTime 0
+        eraHistory = eraHistoryWithHorizonAt 15
+        timeHandle = mkTimeHandle now systemStart eraHistory
+        slotInside = SlotNo 14
+        converted = slotToUTCTime timeHandle slotInside
+        expected = Right $ posixSecondsToUTCTime 14
+    converted `shouldBe` expected
```

Comment:
Could: Make this a property and use `genTimeParams` and `mkTimeHandle` with expectations that all slots past horizon will be returning a `Left` etc.

Reply:
Ok I added a todo about converting `TimeHandleSpec` to proper property test.